mitmproxy/controller.py

```diff
@@ -147,23 +167,63 @@ def __call__(self, msg=False):
 NO_REPLY = object()
 
 
-class Reply(object):
+def handler(f):
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        if len(args) == 1:
```

Comment:
Yes, this deals with inheritance, and making sure we can over-ride handlers without ack-ing twice. I'll add a comment.